### ISSUE TITLE
Wrong explanation in the warning W1638? range-builtin-not-iterating

### ISSUE THREAD

**Issue by bluesheeptoken** [PyCQA/pylint#2797]
## Question
In the official documentation, it is stated that 

> range returns an iterator in Python3

https://docs.pylint.org/en/1.6.0/features.html
I agree pretty much about the warning, but I find the explanation False (range does not return an iterator in python3).

This can be check with 

    from collections.abc import Iterator
    isinstance(range(5), Iterator)

Is this suitable for a pull request? Or did I understand wrong?

What is the reason to not stock `range` ?

**Comment by PCManticore** [comment]
Hey @bluesheeptoken Yes, I think it's fine to update that particular message, so feel free to send up a PR! I don't quite get what you mean by not stocking `range`? For context this message is emitted by the Python 3 porting checker when the `range()` builtin is not used in an *iterating* context, which was the general case for Python 2, where `range()` returned a list and thus supported various operations on it such as indexing and slicing.
Also going to close this issue, but feel free to send a PR to change that wording.

**Comment by bluesheeptoken** [comment]
Thank you @PCManticore 
My question was "is it a good practice to assign a variable with a range"
such as 

```python
r = range(5)
for i in r: pass
for i in r: pass
```

? If not why?

I will do the PR !

**Comment by juanarrivillaga** [comment]
This warning doesn't make any sense. `range`, in Python 2 and 3, has always returned a *container*. It has many use-cases outside of iterating.

**Comment by bluesheeptoken** [comment]
`[range](https://docs.python.org/2/library/functions.html#range)` returns a list in Python2, and a `Range` in Python3.
This warning occurs when you want to migrate from Python 2 to Python 3. To warn you these behave differently.
